scene/3d/physics_body_3d.cpp

```diff
@@ -1171,7 +1167,7 @@ bool CharacterBody3D::move_and_slide() {
 
 		sliding_enabled = true;
 
-		if (!found_collision || motion == Vector3()) {
+		if (!collided || motion == Vector3()) {
```

Comment:
Looks perfect only here where you should replace the test by `motion.is_equal_approx(Vector3())`.

(you can also make the same change at the line `1112` and `1175`)

Reply:
Thanks! I've changed the check to `is_equal_approx` in the 3 different places.